lib/formatters/tapFormatter.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+/**
+ * @type {import('stylelint').Formatter}
+ */
+const tapFormatter = (results) => {
+	let lines = [`TAP version 13\n1..${results.length}`];
+
+	results.forEach((result, index) => {
+		lines.push(
+			`${result.errored ? 'not ok' : 'ok'} ${index + 1} - ${result.ignored ? 'ignored ' : ''}${
+				result.source
+			}`,
+		);
+
+		if (result.warnings.length > 0) {
+			lines.push(...['---', 'messages:']);
+
+			result.warnings.forEach((warning) => {
+				lines.push(
+					...[
+						` - message: "${warning.text}"`,
+						`   severity: ${warning.severity}`,
+						`   data:`,
+						`     line: ${warning.line}`,
+						`     column: ${warning.column}`,
+						`     ruleId: ${warning.rule}`,
+					],
+				);
+			});
+
+			lines.push('---');
+		}
+	});
+
+	lines.push('');
```

Comment:
It's not clear to me why this needs to exist, but with some manual testing in another project if this is removed the last result will not be reported for larger sets of results. Would definitely be interested if there's a better approach based on what happens downstream of the formatter.

Reply:
maybe for some tools `\n` is required in the end

Reply:
That makes sense, leaving as-is